packages/replay-next/components/search-files/SearchFiles.tsx

```diff
@@ -2,57 +2,107 @@ import {
 
+function FilterButton({
+  icon,
+  toggle,
+  active,
+  tooltip: tooltipTitle,
+  ...props
+}: React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  icon: IconType;
+  toggle: () => any;
+  active: boolean;
+  tooltip: string;
+}) {
+  const { onMouseEnter, onMouseLeave, tooltip } = useTooltip({
+    position: "below",
+    tooltip: tooltipTitle,
+  });
+
+  return (
+    <>
+      <button
+        {...props}
+        data-active={active}
+        onMouseEnter={onMouseEnter}
+        onMouseLeave={onMouseLeave}
+        onClick={() => toggle()}
+        className={active ? styles.SelectedSearchFilterButton : styles.SearchFilterButton}
+      >
+        <Icon className={styles.SearchFilterIcon} type={icon} />
+      </button>
+      {tooltip}
+    </>
+  );
+}
+
 export default function SearchFiles({ limit }: { limit?: number }) {
   const client = useContext(ReplayClientContext);
+  const { openSourceIds } = useContext(SourcesContext);
 
-  const [includeNodeModules, setIncludeNodeModules] = useState(false);
+  const [excludeNodeModules, setExcludeNodeModules] = useState(true);
   const [queryForDisplay, setQueryForDisplay] = useState("");
-  const [queryForSuspense, setQueryForSuspense] = useState("");
+  const [includedFiles, setIncludedFiles] = useState("");
+  const [excludedFiles, setExcludedFiles] = useState("");
+  const [openEditors, setOpenEditors] = useState(false);
+  const [useRegex, setUseRegex] = useState(false);
+  const [caseSensitive, setCaseSensitive] = useState(false);
+  const [wholeWord, setWholeWord] = useState(false);
+  const queryForSearch = useDebounce(queryForDisplay, 500);
   const [, dismissSearchSourceTextNag] = useNag(Nag.SEARCH_SOURCE_TEXT);
+  const isPending = queryForSearch !== queryForDisplay;
 
   const inputRef = useRef<HTMLInputElement>(null);
 
-  const streaming = searchCache.stream(client, queryForSuspense, includeNodeModules, limit);
+  const streaming = searchCache.stream(
+    client,
+    queryForSearch,
+    excludeNodeModules,
+    includedFiles,
+    excludedFiles,
+    openEditors ? openSourceIds : null,
+    useRegex,
+    caseSensitive,
+    wholeWord,
+    limit
+  );
 
   const { status } = useStreamingValue(streaming);
 
   const didError = status === STATUS_REJECTED;
 
-  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setQueryForDisplay(event.target.value);
-  };
+  const onChange =
+    (dispatcher: React.Dispatch<React.SetStateAction<string>>) =>
+    (event: ChangeEvent<HTMLInputElement>) => {
+      dispatcher(event.target.value);
+    };
 
   const onKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
     switch (event.key) {
-      case "Enter":
-      case "NumpadEnter":
-        startTransition(() => {
-          setQueryForSuspense(queryForDisplay);
-        });
-        break;
       case "Escape":
-        setQueryForDisplay(queryForSuspense);
+        setQueryForDisplay(queryForSearch);
         break;
     }
```

Comment:
I think we should just remove this handler entirely. This "escape to cancel" behavior doesn't make sense anymore if we aren't explicitly applying the text with "Enter"

Reply:
I agree. I have pushed these changes. Could you take a look? Should be good to merge